src/filters.php

```diff
@@ -32,11 +33,62 @@
 });
 
 /**
- * Use theme wrapper
+ * Use Blade template engine
  */
-add_filter('template_include', function ($main) {
-    if (!is_string($main) && !(is_object($main) && method_exists($main, '__toString'))) {
-        return $main;
+array_map(function ($tag) {
+    add_filter("{$tag}_template_hierarchy", function($templates) {
+        return array_merge(str_replace('.php', '.blade.php', $templates), $templates);
+    });
+}, [
+    'index', '404', 'archive', 'author', 'category', 'tag', 'taxonomy', 'date', 'home',
+    'front_page', 'page', 'paged', 'search', 'single', 'singular', 'attachment'
+]);
+add_filter('template_include', function ($template) {
+    $blade_template = (!strpos($template, '.blade.php')) ? str_replace('.php', '.blade.php', $template) : $template;
+    $blade_template = locate_template(basename($blade_template));
+
+    if (!file_exists($blade_template)) {
+        return $template;
+    }
+
+    $container = Container::getInstance();
```

Comment:
I can understand the reasoning on this, making sure everything that's used is in the same namespace. However, I was also considering that really, the Container class should be global and only one instance of, just like in a Laravel application. The IoC container is something that can be used anywhere and give you instances of anything you need.

Basically, I don't know if we necessarily want to make that Roots specific. I don't think it would necessarily be bad if other plugins called `Illuminate\Container\Container` directly. On the other hand, perhaps we do move it out to it's own project along with the Blade service provider, and also add some things that are used in the global scope of WP, like `$wpdb`, and add that to the container, it's a container that could be used across plugins and themes in general.